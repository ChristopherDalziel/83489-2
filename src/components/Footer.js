import React from "react";
import styled from "@emotion/styled";

const FooterDiv = styled.footer`
  margin-top: auto;
  height: 6vh;
  background-color: #e63846;
  border-radius: 50px 0px 0px 0px;
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
  padding: 2vh;
  font-size: 50px;
`;

const Footer = () => {
  return (
    <FooterDiv>
      <a
        target="_blank"
        href="https://github.com/ChristopherDalziel/83489-2"
        rel="noopener noreferrer"
        class="fab fa-github-square socials"
        id="footerLink"
      >
        <span>Link to projects GitHub repository</span>
      </a>
    </FooterDiv>
  );
};

export default Footer;
