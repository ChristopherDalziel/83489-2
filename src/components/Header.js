import React from "react";
import styled from "@emotion/styled";
import logo from "../assets/logo.png";

const HeaderDiv = styled.header`
  padding: 0;
  margin: 0;
  background-color: #e63846;
  height: 11vh;
  border-radius: 0px 0px 50px 0px;
  box-shadow: 17px 14px 17px -11px rgba(0, 0, 0, 0.2);
  @media (max-width: 815px) {
    border-radius: 0px;
    font-size: 25px;
    height: 10vh;
  }
`;

const Title = styled.h1`
  margin: 0 0 5px 0;
  padding: 0px 10px 0px 10px;
  width: 30vw;
  @media (max-width: 815px) {
    margin-bottom: 5px;
    font-size: 25px;
    width: 75vw;
  }
`;

const SubTitle = styled.h2`
  margin: 0;
  font-size: 1rem;
  width: 20.75vw;
  background-color: black;
  padding: 5px 10px 5px 10px;
  border-radius: 0px 50px 50px 0px;
  @media (max-width: 815px) {
    padding: 5px
    border-radius: 0px;
    font-size: 12px;
    width: 100vw;
  }
`;

const LogoSmall = styled.img`
  height: 90px;
  float: right;
  transition: transform 0.2s;
  :hover {
    -ms-transform: scale(0.9);
    -webkit-transform: scale(0.9);
    transform: scale(0.9);
  }
  @media (max-width: 815px) {
    visibility: hidden;
  }
`;

const Header = (props) => {
  return (
    <HeaderDiv>
      <Title>
        {props.title}
        <LogoSmall src={logo} alt="Site logo" />
      </Title>

      {props.subTitle && <SubTitle>{props.subTitle}</SubTitle>}
    </HeaderDiv>
  );
};

Header.defaultProps = {
  title: "Shape Sorter Online",
  subTitle: "Leave your shape sorting to a computer",
};

export default Header;
