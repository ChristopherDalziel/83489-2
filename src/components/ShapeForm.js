import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import {
  sortByTriangles,
  sortByCircles,
  sortByQuadrilaterals,
} from "../actions/filters";

const FormContainer = styled.div`
  padding: 20px 10px 0px 10px;
  color: black;
  @media (max-width: 815px) {
    width: 50vw;
  }
`;

const Instructions = styled.p`
  width: 50vw;
`;

export class ShapeForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sideOne: props.shape ? props.shape.sideOne : "",
      sideTwo: props.shape ? props.shape.sideTwo : "",
      sideThree: props.shape ? props.shape.sideThree : "",
      sideFour: props.shape ? props.shape.sideFour : "",
      error: "",
    };
  }

  onSideOneChange = (e) => {
    const sideOne = e.target.value;
    if (!sideOne || sideOne.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => {
        return { sideOne };
      });
    }
  };

  onSideTwoChange = (e) => {
    const sideTwo = e.target.value;
    if (!sideTwo || sideTwo.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => {
        return { sideTwo };
      });
    }
  };

  onSideThreeChange = (e) => {
    const sideThree = e.target.value;
    if (!sideThree || sideThree.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => {
        return { sideThree };
      });
    }
  };

  onSideFourChange = (e) => {
    const sideFour = e.target.value;

    if (!sideFour || sideFour.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => {
        return { sideFour };
      });
    }
  };

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.sideOne) {
      this.setState(() => ({
        error: "Please provide at least 'Side One' ",
      }));
    } else if (
      this.state.sideOne &&
      this.state.sideTwo &&
      !this.state.sideThree
    ) {
      this.setState(() => ({
        error: "Please provide a shape with more or less than 2 sides",
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        sideOne: this.state.sideOne,
        sideTwo: this.state.sideTwo,
        sideThree: this.state.sideThree,
        sideFour: this.state.sideFour,
      });

      if (
        this.state.sideOne &&
        this.state.sideTwo &&
        this.state.sideThree &&
        this.state.sideFour
      ) {
        this.props.sortByQuadrilaterals();
      } else if (
        this.state.sideOne &&
        this.state.sideTwo &&
        this.state.sideThree
      ) {
        this.props.sortByTriangles();
      } else if (this.state.sideOne) {
        this.props.sortByCircles();
      }
    }
  };

  render() {
    return (
      <FormContainer>
        <Instructions
          aria-readonly="true"
          role="Instructions on how to use this website"
        >
          Please enter your the lengths of each shape of the shape within the
          fields below, consider the lengths are been measured in pixels so you
          may want to use double digits to ensure your shape is visible!
          <Instructions>
            Then I will return your shape below, sorted into it's category!
          </Instructions>
        </Instructions>
        <form onSubmit={this.onSubmit}>
          {this.state.error && <p>{this.state.error}</p>}
          <label for="Shape Side One">Side One:</label>
          <input
            autoFocus
            onChange={this.onSideOneChange}
            value={this.state.sideOne}
            placeholder="Side One"
          ></input>
          <label for="Shape Side Two">Side Two:</label>
          <input
            disabled={!this.state.sideOne ? true : false}
            onChange={this.onSideTwoChange}
            value={this.state.sideTwo}
            placeholder="Side Two"
          ></input>
          <label for="Shape Side Three">Side Three:</label>
          <input
            disabled={!this.state.sideTwo ? true : false}
            onChange={this.onSideThreeChange}
            value={this.state.sideThree}
            placeholder="Side Three"
          ></input>
          <label for="Shape Side Four">Side Four:</label>
          <input
            disabled={!this.state.sideThree ? true : false}
            onChange={this.onSideFourChange}
            value={this.state.sideFour}
            placeholder="Side Four"
          ></input>
          <button>Create new shape</button>
        </form>
      </FormContainer>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    sortByTriangles: () => dispatch(sortByTriangles()),
    sortByCircles: () => dispatch(sortByCircles()),
    sortByQuadrilaterals: () => dispatch(sortByQuadrilaterals()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ShapeForm);
