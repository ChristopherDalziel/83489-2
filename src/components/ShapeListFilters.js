import React from "react";
import { connect } from "react-redux";
import styled from "@emotion/styled";
import {
  sortByTriangles,
  sortByCircles,
  sortByQuadrilaterals,
} from "../actions/filters";

const Body = styled.div`
  margin: 10px;
`;

class ShapeListFilters extends React.Component {
  onSortChange = (e) => {
    if (e.target.value === "Triangles") {
      return this.props.sortByTriangles();
    } else if (e.target.value === "Circles") {
      return this.props.sortByCircles();
    } else if (e.target.value === "Quadrilaterals") {
      return this.props.sortByQuadrilaterals();
    }
  };

  render() {
    return (
      <Body>
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="Triangles">Triangles</option>
          <option value="Circles">Circles</option>
          <option value="Quadrilaterals">Quadrilaterals</option>
        </select>
      </Body>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    sortByTriangles: () => dispatch(sortByTriangles()),
    sortByCircles: () => dispatch(sortByCircles()),
    sortByQuadrilaterals: () => dispatch(sortByQuadrilaterals()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ShapeListFilters);
