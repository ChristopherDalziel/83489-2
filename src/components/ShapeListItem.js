import React from "react";
import styled from "@emotion/styled";
import { Stage, Layer, Shape } from "react-konva";

// import GonnaBeMyShape from "../components/Shape";

const ShapeListItem = ({ sideOne, sideTwo, sideThree, sideFour }) => {
  const Quadrilateral = () => {
    console.log(sideOne);
    return (
      <Stage width={window.innerWidth} height={window.innerHeight}>
        <Layer>
          <Shape
            sceneFunc={(context, shape) => {
              context.beginPath();
              context.moveTo(100, 100); // placement within the canvas
              context.lineTo(`${sideOne * 5}`, 100); // 1st value = top line
              context.lineTo(`${sideThree * 5}`, `${sideTwo * 5}`); // 1st value = bottom line, 2nd value = right line
              context.lineTo(100, `${sideFour * 5}`); // 2nd value = left line
              context.lineTo(100, 100); // close shape
              context.closePath();
              // (!) Konva specific method, it is very important
              context.fillStrokeShape(shape);
            }}
            fill="#00D2FF"
            stroke="black"
            strokeWidth={4}
          />
        </Layer>
      </Stage>
    );
  };
  const Circle = styled.div`
    margin: 20px;
    width: ${sideOne * 10}px;
    height: ${sideOne * 10}px;
    background: black;
    border-radius: 50%;
  `;

  const Triangle = styled.div`
    margin: 20px;
    width: 0;
    height: 0;
    border-bottom: ${sideOne * 10}px solid black;
    border-left: ${sideTwo * 10}px solid transparent;
    border-right: ${sideThree * 10}px solid transparent;
  `;

  // const Quadrilateral = styled.div`
  //   margin: 20px;
  //   width: ${sideFour * 10}px;
  //   height: 0;
  //   border-bottom: ${sideOne * 10}px solid black;
  //   border-left: ${sideTwo * 10}px solid transparent;
  //   border-right: ${sideThree * 10}px solid transparent;
  // `;

  // const Square = styled.div`
  //   height: ${sideOne * 10}px;
  //   width: ${sideOne * 10}px;
  //   background: black;
  // `;

  return (
    <div>
      <div>
        {sideOne && !sideTwo && <Circle />}
        {sideTwo && sideTwo && sideThree && !sideFour && <Triangle />}
        {sideTwo && sideTwo && sideThree && sideFour && <Quadrilateral />}
        {/* {sideOne && sideTwo && sideThree && sideFour === sideOne && <Square />} */}
      </div>
    </div>
  );
};

export default ShapeListItem;
