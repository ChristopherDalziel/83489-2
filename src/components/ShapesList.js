import React from "react";
import { connect } from "react-redux";

// import ShapeListItem from "./ShapeListItem";
import sortShapes from "../selectors/shapes";

import { Circle, Triangle, Rectangle } from "react-shapes";

const ShapeListItem = ({ sideOne, sideTwo, sideThree, sideFour }) => {
  return (
    <div>
      <p>
        {sideOne && !sideTwo && (
          <Circle
            r={sideOne * 10}
            fill={{ color: "#2409ba" }}
            stroke={{ color: "#E65243" }}
            strokeWidth={3}
          />
        )}
        {sideTwo && sideTwo && sideThree && !sideFour && (
          <Triangle
            width={sideOne * 10}
            height={sideTwo * 10}
            fill={{ color: "#2409ba" }}
            stroke={{ color: "#E65243" }}
            strokeWidth={3}
          />
        )}
        {sideTwo && sideTwo && sideThree && sideFour && (
          <Rectangle
            width={sideOne * 10}
            height={sideTwo * 10}
            fill={{ color: "#2409ba" }}
            stroke={{ color: "#E65243" }}
            strokeWidth={3}
          />
        )}
      </p>
    </div>
  );
};

const ShapeList = (props) => {
  return (
    <div>
      <>
        <h1>Sorted Shapes</h1>
        {props.shapes.map((shape) => {
          return <ShapeListItem {...shape} />;
        })}
      </>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    shapes: sortShapes(state.shapes, state.filters),
  };
};

export default connect(mapStateToProps)(ShapeList);
