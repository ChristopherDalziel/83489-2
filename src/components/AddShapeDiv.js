import React from "react";
import { connect } from "react-redux";
import styled from "@emotion/styled";
import ShapeForm from "./ShapeForm";
import { addShape } from "../actions/shapes";

const Button = styled.button`
  margin: 10px 0px 0px 10px;
  width: 10vw;
  @media (max-width: 815px) {
    width: 50vw;
    text-align: left;
  }
`;

export class AddShapeDiv extends React.Component {
  state = {
    menuShown: false,
  };

  onSubmit = (shape) => {
    this.props.addShape(shape);
    this.setState({ menuShown: false });
  };

  render() {
    return (
      <>
        {this.state.menuShown ? (
          <Button
            type="toggle"
            onClick={() => this.setState({ menuShown: !this.state.menuShown })}
          >
            Cancel
          </Button>
        ) : (
          <Button
            type="toggle"
            onClick={() => this.setState({ menuShown: !this.state.menuShown })}
          >
            Create your own shape
          </Button>
        )}

        {this.state.menuShown ? <ShapeForm onSubmit={this.onSubmit} /> : ""}
      </>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addShape: (shape) => dispatch(addShape(shape)),
  };
};

export default connect(undefined, mapDispatchToProps)(AddShapeDiv);
