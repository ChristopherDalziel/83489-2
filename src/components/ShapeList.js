import React from "react";
import { connect } from "react-redux";
import styled from "@emotion/styled";
import ShapeListItem from "./ShapeListItem";
import sortShapes from "../selectors/shapes";

const Body = styled.div`
  margin: 10px;
  overflow: scroll;
`;

const H2 = styled.h2`
  color: #e63846;
`;

const NoShapesMessage = styled.p`
  color: #e63846;
`;

export const ShapeList = (props) => {
  return (
    <Body>
      {props.shapes.length === 0 ? (
        <NoShapesMessage>No shapes to display</NoShapesMessage>
      ) : (
        <>
          <H2>Sorted Shapes</H2>
          {props.shapes.map((shape, index) => {
            return <ShapeListItem key={index} {...shape} />;
          })}
        </>
      )}
    </Body>
  );
};

const mapStateToProps = (state) => {
  return {
    shapes: sortShapes(state.shapes, state.filters),
  };
};

export default connect(mapStateToProps)(ShapeList);
