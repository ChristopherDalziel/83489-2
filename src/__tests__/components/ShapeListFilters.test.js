import React from "react";
import { shallow } from "enzyme";
import { ShapeListFilters } from "../../components/ShapeListFilters";
import {
  filters,
  alternateFilterCircles,
  alternateFilterQuadrilaterals,
} from "../fixtures/filters";

import toJSON from "enzyme-to-json";

let sortByTriangles, sortByCircles, sortByQuadrilaterals, wrapper;

beforeEach(() => {
  sortByTriangles = jest.fn();
  sortByCircles = jest.fn();
  sortByQuadrilaterals = jest.fn();
  wrapper = shallow(
    <ShapeListFilters
      filters={filters}
      sortByCircles={sortByCircles}
      sortByQuadrilaterals={sortByQuadrilaterals}
      sortByTriangles={sortByTriangles}
    />
  );
});

test("Should render ShapeListFilters with default filter correctly", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should render ShapeListFilters with alternateFilterCircle data correctly", () => {
  wrapper.setProps({ filters: alternateFilterCircles });
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should render ShapeListFilters with alternateFilterQuadrilaterals data correctly", () => {
  wrapper.setProps({ filters: alternateFilterQuadrilaterals });
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should sort by Triangles", () => {
  const value = "Triangles";
  wrapper.find("select").simulate("change", { target: { value } });
  expect(sortByTriangles).toHaveBeenCalled();
});

test("Should sort by Circles", () => {
  const value = "Circles";
  wrapper.find("select").simulate("change", { target: { value } });
  expect(sortByCircles).toHaveBeenCalled();
});

test("Should sort by Quadrilaterals", () => {
  const value = "Quadrilaterals";
  wrapper.find("select").simulate("change", { target: { value } });
  expect(sortByQuadrilaterals).toHaveBeenCalled();
});
