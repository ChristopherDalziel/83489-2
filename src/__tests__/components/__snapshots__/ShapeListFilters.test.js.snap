// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ShapeListFilters with alternateFilterCircle data correctly 1`] = `
<Styled(div)>
  <select
    onChange={[Function]}
    value="Circles"
  >
    <option
      value="Triangles"
    >
      Triangles
    </option>
    <option
      value="Circles"
    >
      Circles
    </option>
    <option
      value="Quadrilaterals"
    >
      Quadrilaterals
    </option>
  </select>
</Styled(div)>
`;

exports[`Should render ShapeListFilters with alternateFilterQuadrilaterals data correctly 1`] = `
<Styled(div)>
  <select
    onChange={[Function]}
    value="Quadrilaterals"
  >
    <option
      value="Triangles"
    >
      Triangles
    </option>
    <option
      value="Circles"
    >
      Circles
    </option>
    <option
      value="Quadrilaterals"
    >
      Quadrilaterals
    </option>
  </select>
</Styled(div)>
`;

exports[`Should render ShapeListFilters with default filter correctly 1`] = `
<Styled(div)>
  <select
    onChange={[Function]}
    value="Triangles"
  >
    <option
      value="Triangles"
    >
      Triangles
    </option>
    <option
      value="Circles"
    >
      Circles
    </option>
    <option
      value="Quadrilaterals"
    >
      Quadrilaterals
    </option>
  </select>
</Styled(div)>
`;
